### ISSUE TITLE
US-2.2 - Train ResNet50 Model

### ISSUE THREAD

**Issue by malakzaidi** [malakzaidi/medical-XAI-skin-lesion-classification#17]
**As a ML Engineer, I want to train ResNet50 to achieve >75% validation accuracy.**
**Acceptance Criteria:**
- [ ] Training script: backend/models/train_resnet50.py
- [ ] Training loop implemented with progress bars
- [ ] Validation loop implemented
- [ ] Best model checkpoint saved
- [ ] Training history logged (loss, accuracy)
- [ ] Validation accuracy >75%
- [ ] Learning rate scheduler implemented
- [ ] Early stopping (optional)
- [ ] Training curves saved as graph
**Hyperparameters:**
- Optimizer: Adam
- Learning rate: 0.0001
- Batch size: 32
- Epochs: 10-15
- Loss: CrossEntropyLoss
- Scheduler: ReduceLROnPlateau
**Estimated Time:** 2h15 (includes training time)
